### DISCUSSION TITLE
StreamedResponse close connexion server side and stop reconnecting client side

### DISCUSSION THREAD

**Discussion by thgauthier** [symfony/symfony#49933] [Q&A] [answered]
Hi !
How send a HTTP 204 response code inside the callBack StreamedResponse function ?
I need to do that in order to stop client  reconnecting [server-sent-events-intro](https://html.spec.whatwg.org/multipage/server-sent-events.html#server-sent-events-intro), rather than stopped from client side with a if statement.

![code](https://user-images.githubusercontent.com/95478233/229926935-a3a31952-7423-4b84-8112-1401fc8d3f68.png)
Thanks

**Comment by ghost** [comment]
StreamedResponse extends Response class. You can pass a status code to the constructor or use setStatusCode().
However 204 means No Content and streamed response returns content so unexpected behavior may occur (like content may be removed, Content-Type|Content-Length as well) so please be aware. But if this will solve your problem then OK.

**Reply by thgauthier** [reply]
Thank you for your answer.
According to the doc : `Event stream requests can be redirected using HTTP 301 and 307 redirects as with normal HTTP requests. Clients will reconnect if the connection is closed; a client can be told to stop reconnecting using the HTTP 204 No Content response code.`
But yes I tried setStatusCode() inside the callback using the $response passing inside the callBack with `function ()  use ($response){`
But with no effect.

**Reply by ghost** [reply]
> But yes I tried setStatusCode() inside the callback using the $response passing inside the callBack

status code is sent before content. If you set (i.e. change) status inside a content callback then it's too late, it will send the default code i.e. 200. Set it after line 14.

**Reply by stefangr** [reply]
You can only use the 204 response in your controller if the controller knows there is no data to be sent.

If a client has an active connection that received data already, the server side needs to inform the client side there is no more data by using a different event type.
The client side then has to close the connection by stop listening to the events (invoking the close method on the EventSource instance).

**Reply by thgauthier** [reply, marked as answer]
OK Thank you very much !
